skimpdf: init at 1.7.2

YvesStraten opened the pull request:
## Description of changes
Init skimpdf in nixpkgs
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

YvesStraten committed:
maintainers: add YvesStraten

YvesStraten committed:
skimpdf: init at 1.7.2

quantenzitrone commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,34 @@
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
+    maintainers = with maintainers; [ YvesStraten ];
+    platforms = [ "x86_64-darwin" "aarch64-darwin" ];
```
```suggestion
    platforms = platforms.darwin;
```

This corresponds to `[ "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]`.

YvesStraten replied:
Understood, adding changes

raspher commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,34 @@
+  sourceRoot = ".";
+
+  installPhase = ''
+    mkdir -p $out/Applications
+    cp -R *.app $out/Applications
```
```suggestion
    runHook preInstall
    mkdir -p $out/Applications
    cp -R *.app $out/Applications
    runHook postInstall
```
https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

raspher commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,34 @@
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
+    maintainers = with maintainers; [ YvesStraten ];
+    platforms = platforms.darwin;
```
afaik this is not builded from source, so please add `sourceProvenance`
https://nixos.org/manual/nixpkgs/stable/#sec-meta-sourceProvenance

raspher commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,34 @@
+    description = "Skim is a PDF reader and note-taker for OS X";
+    homepage = "https://skim-app.sourceforge.io/";
+    license = licenses.bsd0;
+    mainProgram = "Skim.app";
```
If there is one `mainProgram` you shouldn't copy multiple binaries to $out (`cp -R *.app $out/Applications` in installPhase)

If there are many executables, this `meta.mainProgram` should not be defined

https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

raspher replied:
Still relevant

YvesStraten replied:
There is only Skim.app so the mainProgram tag should be fine

YvesStraten replied:
Oh wait sorry, for some reason my changes to the installPhase did not get synced, fixing it right now

YvesStraten replied:
Has been done

YvesStraten commented:
Hello, @raspher, thank you very much for your suggestions. I have added them to the derivations and squashed the commits accordingly. This should hopefully be everything for this package. Thank you very much and have a very nice day/night 🙏

raspher commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,37 @@
+, lib
+, undmg
+, fetchurl
+,
```
```suggestion
```

raspher commented on pkgs/by-name/sk/skimpdf/package.nix:
```diff
@@ -0,0 +1,37 @@
+}:
+stdenv.mkDerivation rec {
+  pname = "Skim";
+  version = "1.6.21";
```
new version released https://sourceforge.net/projects/skim-app/files/Skim/

YvesStraten replied:
The new version will be added in a new pr, like nixpkgs wants it to be

YvesStraten commented:
All suggestions have been added. Thank you very much for your suggestions

YvesStraten commented:
Everything should be correct and this can be hopefully merged

RossComputerGuy reviewed (approved):
I approb

tomberek reviewed (approved):
Thanks for the contribution!

YvesStraten commented:
No problem, thank you so much for approving and merging 🙏
